File: latte/src/main/java/gg/beemo/latte/broker/rpc/RpcStatus.kt
package gg.beemo.latte.broker.rpc

class RpcStatus(val code: Int) {

    companion object {
        val OK = RpcStatus(0)
        val UNKNOWN = RpcStatus(999_999)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is RpcStatus) return false
        return code == other.code
    }

    override fun hashCode(): Int {
        return code
    }

}

File: water/src/broker/rpc/RpcStatus.ts
export class RpcStatus {
	public static readonly OK = new RpcStatus(0);
	public static readonly UNKNOWN = new RpcStatus(999_999);

	public constructor(public readonly code: number) {}

	public equals(other: RpcStatus): boolean {
		return this.code === other.code;
	}

	public toString(): string {
		return `RpcStatus(${this.code})`;
	}
}
